pkg/controller/sparkapplication/controller.go

```diff
@@ -534,33 +533,47 @@ func (c *Controller) syncSparkApplication(key string) error {
 			appToUpdate = c.submitSparkApplication(appToUpdate)
 		}
 	case v1beta2.PendingSubmissionState:
-		// Check the status of the submission Job and set the application status accordingly.
-		succeeded, completionTime, err := c.subJobManager.hasJobSucceeded(appToUpdate)
-
-		if succeeded != nil {
-			// Submission Job terminated in either success or failure.
-			if *succeeded {
-				c.createSparkUIResources(appToUpdate)
-				appToUpdate.Status.AppState.State = v1beta2.SubmittedState
-				appToUpdate.Status.ExecutionAttempts++
-				if completionTime != nil {
-					appToUpdate.Status.SubmissionTime = *completionTime
-				}
-				c.recordSparkApplicationEvent(appToUpdate)
+		//Resubmission is based on resource quota. We wait and then see if the interval passed to rerun
+		if app.Spec.Mode == v1beta2.ClientMode {
+			glog.Info("in pending submission")
+			var interval int64 = 257
+			time.Sleep(time.Duration(interval) * time.Second)
```

Comment:
Let's not sleep here . This will just block the worker thread. Have a look at how the operator does for the case when submit is run within the operator which should be very similar to this case:
https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/master/pkg/controller/sparkapplication/controller.go#L556

Reply:
I used the same method right below its just renamed. Without sleep checking if it is time to retry will fail without trying again. I'll remove the sleep and see how the controller can pick it back up